int32 id
int32 nChannels 
int32 params
channel[] channels